bundles/org.openhab.binding.enphase/src/main/java/org/openhab/binding/enphase/internal/handler/EnvoyConnectorWrapper.java

```diff
@@ -0,0 +1,174 @@
+    private final HttpClient httpClient;
+    private @Nullable EnvoyConnector connector;
+    private @Nullable String version;
+
+    public EnvoyConnectorWrapper(final HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    public @Nullable String getVersion() {
+        return version;
+    }
+
+    /**
+     * Sets the Envoy software version as retreived from the Envoy data.
+     *
+     * @param version Envoy software version.
+     */
+    public void setVersion(final @Nullable String version) {
+        if (version == null) {
+            return;
+        }
+        logger.debug("Set Envoy version found in the Envoy data: {}", version);
+        this.version = version;
+    }
+
+    /**
+     * Set the connector given the configuration.
+     *
+     * @param configuration configuration to use to set the connector.
+     * @return Returns configuration error message or empty string if no configuration errors present
+     * @throws EnphaseException
+     */
+    public synchronized String setConnector(final EnvoyConfiguration configuration) throws EnphaseException {
+        final EnvoyConnector connector = determineConnector(configuration.hostname);
+        final String message = connector.setConfiguration(configuration);
+
+        // Only set connector if no error messages.
+        if (message.isEmpty()) {
+            this.connector = connector;
+        }
+        return message;
+    }
+
+    /**
+     * @return Returns true if a connection with the Envoy has been established.
+     */
+    public boolean hasConnection() {
+        return connector != null;
+    }
+
+    /**
+     * @return Returns the connector when present. This method should only be called when a connector is present as
+     *         returned by the {@link #hasConnection()} method.
+     */
+    public EnvoyConnector getConnector() throws EnvoyConnectionException {
+        final EnvoyConnector connector = this.connector;
+
+        if (connector == null) {
+            throw new EnvoyConnectionException("No connection to the Envoy. Check your configuration.");
+        }
+        return connector;
+    }
+
+    private EnvoyConnector determineConnector(final String hostname) throws EnphaseException {
+        final EnvoyConnector connectorByVersion = determineConnectorOnVersion();
+
+        if (connectorByVersion != null) {
+            return connectorByVersion;
+        }
+
+        if (hostname.isBlank()) {
+            throw new EnvoyNoHostnameException("No hostname available.");
+        }
+        final EnvoyConnector envoyConnector = new EnvoyConnector(httpClient);
+        final String version = envoyConnector.checkConnection(hostname);
+
+        if (version != null) {
+            this.version = version;
+            final int majorVersionNumber = determineMajorVersionNumber();
+
+            if (majorVersionNumber > 0 && majorVersionNumber < 7) {
+                logger.info(
```

Comment:
Perhaps **debug** would be sufficient here, since the user probably doesn't need to see this every time?

Reply:
Changed to debug